server-spi-private/src/main/java/org/keycloak/useragent/UserAgentProvider.java

```diff
@@ -0,0 +1,13 @@
+package org.keycloak.useragent;
+
+import nl.basjes.parse.useragent.UserAgentAnalyzer;
+import org.keycloak.provider.Provider;
+
+public interface UserAgentProvider extends Provider {
+
+    UserAgentAnalyzer userAgentAnalyser();
```

Comment:
Is it possible that this method is changed to something like `DeviceRepresentation getDeviceRepresentation()` ?

As it will be nice that SPI is generic and does not use any YAUAA specific stuff.

That will allow that for example people can plug their own impl based on completely different library if they want to.